students/Dennis_Coffey/lesson07/html_render.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov 29 21:13:15 2018
+
+@author: dennis
+"""
+
+"""HTML renderer for Lesson 07"""
+
+# Base element class
+class Element():
+    tag = '' # Element type
+    indent_size = ' '* 4 # Indent of tag
+#    indent_size = ''* 4 # Use this cause no indent for Steps 1-8
+    inline = False # Defines whether content is to be displayed on same line
+    empty = False  # Defines whether tag is an empty tag
+    def __init__(self, content=None, **kwargs):
+        self.content_list = []
+        self.kwargs = kwargs
+        if content:
+            self.content_list.append(content)
+        
+    # Append content
+    def append(self, content):
+        self.content_list.append(content)
+        
+    # Start tag for element
+    def starttag(self, tag, indent):
+        if tag:
+            tag = self.indent_size * indent + '<' + tag
+            if self.kwargs:
+                attribute = ''
+                for k in self.kwargs:
+                    attribute += ' ' + k + '=\"' + self.kwargs[k] + '\"'
+                tag += attribute
+            tag += '>'
+        return tag
+        
+    # End tag for element
+    def endtag(self, tag, indent):
+        if tag and not self.inline:
+            tag = self.indent_size * indent + '</' + tag + '>'
+        elif tag:
+            tag = '</' + tag + '>'
+        return tag
+        
+    # Generate start tag, content and closing tag
+    def render(self, output_file, indent=0):
+        # Render opening tag
+        output_file.write(self.starttag(self.tag, indent))
+        if not self.inline and not self.empty:
+            output_file.write('\n')
+        
+        # Render content
+        if self.content_list:
```

Comment:
no need to check this, your for loop below will just attempt to iterate over empty list and nothing will happen